.github/workflows/smoke.yaml

```diff
@@ -0,0 +1,43 @@
+# This github workflow is for smoke testing the latest release of the actions defined in this
+# repo. This workflow is executed whenever this file is changed. That will happen when the "uses"
+# line below is updated to reference the latest release as defined in the release process.
+
+name: "Release Smoke Test"
+on:
+  push:
+    paths:
+      - '.github/workflows/smoke.yaml'
+jobs:
+  release_smoke:
+    strategy:
+      matrix:
+        clusters:
+         - distribution: GKE
+           version: "1.23"
+
+         - distribution: Kubeception
+           version: "1.22"
+    runs-on: ubuntu-latest
+    env:
+      GOOGLE_APPLICATION_CREDENTIALS: gac.json
+    steps:
+      - run: |
+          cat <<'EOF' >> gac.json
+          ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+          EOF
+
+      - name: Kubectl tool installer
+        uses: Azure/setup-kubectl@v3
+
+      # The provision-cluster action will automatically register a cleanup hook to remove the
+      # cluster it provisions when the job is done.
+      - uses: datawire/infra-actions/provision-cluster@v0.1.3
```

Comment:
@aosoriodw No, if you look at the commit history you can see that's what I tried first, but unfortunately dynamic stuff is not allowed in the context of the `uses` field because github has certain policy enforcement features which would be difficult or impossible to implement if that dynamicness were allowed.

Reply:
This is the discussion that definitively classifies the behavior as a "feature" not a bug: https://github.com/orgs/community/discussions/25246